public/app/features/plugins/extensions/usePluginComponents.tsx

```diff
@@ -27,7 +28,7 @@ export function usePluginComponents<Props extends object = {}>({
   return useMemo(() => {
     // For backwards compatibility we don't enable restrictions in production or when the hook is used in core Grafana.
     const enableRestrictions = isGrafanaDevMode() && pluginContext;
-    const components: Array<React.ComponentType<Props>> = [];
+    const components: Array<React.ComponentType<Props> & { meta: PluginExtensionComponentMeta }> = [];
```

Comment:
nit: we could reuse the type here from my previous comment, WDYT?

Reply:
Leave as is

Reply:
Okay, lets leave it for now and introduce a type if this starts to be referenced in many places.